src/core/model/Stroke.cpp

```diff
@@ -72,6 +78,68 @@ auto Stroke::cloneStroke() const -> Stroke* {
 
 auto Stroke::clone() -> Element* { return this->cloneStroke(); }
 
+std::unique_ptr<Stroke> Stroke::cloneSection(const PathParameter& lowerBound, const PathParameter& upperBound) const {
+    auto s = std::make_unique<Stroke>();
+    s->applyStyleFrom(this);
+    s->width = this->width;  // stroke width, not bounding box width
+
+    if (upperBound.index == lowerBound.index) {
+        const Point& p = this->getPoint(lowerBound);
+        const Point& q = this->getPoint(upperBound);
+        s->points = {p, q};
+    } else {
+        s->points.reserve(upperBound.index - lowerBound.index + 2);
+
+        // First point
+        s->points.emplace_back(this->getPoint(lowerBound));
+
+        // end point of segment of index lowerBound.index
+        auto it = this->points.cbegin() + (std::ptrdiff_t)lowerBound.index + 1;
+
+        // end point of segment of index upperBound.index
+        auto endIt = this->points.cbegin() + (std::ptrdiff_t)upperBound.index + 1;
+
+        std::copy(it, endIt, std::back_inserter(s->points));
+
+        // Last point
+        if (upperBound.t != 0.0) {
+            s->points.emplace_back(this->getPoint(upperBound));
+        }
+    }
+
+    // Remove unused pressure value
+    s->points.back().z = Point::NO_PRESSURE;
+
+    return s;
+}
+
+std::unique_ptr<Stroke> Stroke::cloneCircularSectionOfClosedStroke(const PathParameter& startParam,
+                                                                   const PathParameter& endParam) const {
+    auto s = std::make_unique<Stroke>();
+    s->applyStyleFrom(this);
+    s->width = this->width;  // stroke width, not bounding box width
+
+    s->points.reserve(this->points.size() - startParam.index + endParam.index + 1);
+
+    // First point
+    s->points.emplace_back(this->getPoint(startParam));
+
+    auto startIt = this->points.cbegin() + (std::ptrdiff_t)startParam.index + 1;
+    assert(startIt != this->points.cend());
+    std::copy(startIt, this->points.cend() - 1, std::back_inserter(s->points));
+
+    auto endIt = this->points.cbegin() + (std::ptrdiff_t)endParam.index + 1;
+    std::copy(this->points.cbegin(), endIt, std::back_inserter(s->points));
+
+    // Last point
+    s->points.emplace_back(this->getPoint(endParam));
+
+    // Remove unused pressure value
+    s->points.back().z = Point::NO_PRESSURE;
```

Comment:
**Nitpick (also the comment above): ** Use std::next, and explain why you have to do something, not what.

For example, why does the startIter >= endIter? I think, this is even clear in this context.